### PR TITLE
Fix infinite loop of setState()

### PR DESCRIPTION
**Pull request by mareklibra**
Prior the fix, the CPU was consumed for 100% and we were loosing event loop processing.

### CODE REVIEW

File: plugins/parodos/src/components/ParodosPage/ParodosPage.tsx [hunk 1]
```diff
@@ -20,11 +20,15 @@ export const ParodosPage: FC<ParodosPageProps> = ({ children, ...props }) => {
   const { pathname } = useLocation();
   const selectedTab = useMemo(
     () =>
-      navigationMap.findIndex(({ routes }) =>
-        routes.find(route => pathname.includes(route)),
+      Math.max(
```
Comment (dagda1) [inline plugins/parodos/src/components/ParodosPage/ParodosPage.tsx:23]: We definitely need this!

File: plugins/parodos/src/components/projectOverview/ProjectOverview.tsx [hunk 2]
```diff
@@ -68,18 +68,22 @@ export const ProjectOverviewPage = (): JSX.Element => {
 
   React.useEffect(() => {
```
Comment (dagda1) [inline plugins/parodos/src/components/projectOverview/ProjectOverview.tsx:69]: I think trying to keep all this in sync will be problematic.

Could we not simplify it by removing `useEffect` and `setFilteredProjects` to something like:

```ts
  const filteredProjects = useMemo(() => {
    if(projectFilter === 'all-projects') {
      return allProjects;
    }

    return allProjects.filter(project => project.status === projectFilter);
  }, [allProjects, projectFilter]);
```

Comment (mareklibra) [reply plugins/parodos/src/components/projectOverview/ProjectOverview.tsx:69]: Right, it makes sense.

However, let's keep the sorting.

**Review by dagda1 — APPROVED** [review]
👍

**Comment by mareklibra** [comment]
Rebased

**Comment by dagda1** [comment]
> Rebased

Apologies, I should have let you merge first.

**Comment by mareklibra** [comment]
> Apologies, I should have let you merge first.

Absolutely no issue, someone needs to be first.

**Pull request merged**